src/hooks/useOnOpenModal.ts

```diff
@@ -0,0 +1,9 @@
+import { useEffect } from "react";
+
+export const useOnOpenModal = (isOpen: boolean, callback: () => void) => {
```

Comment:
What's the reason behind this hook? Maybe we can make something like `useModal` and enable more options to configure?